lib/SIL/Verifier/SILVerifier.cpp

```diff
@@ -792,6 +792,44 @@ static void checkAddressWalkerCanVisitAllTransitiveUses(SILValue address) {
   llvm::report_fatal_error("invoking standard assertion failure");
 }
 
+using VarID = std::tuple<const SILDebugScope *, llvm::StringRef, SILLocation>;
```

Comment:
In theory LLVM already has a DenseMapInfo implementation for std::tuple, that forwards to its members,so it may be enough to just add one for SILLocation (assuming that's the missing one) https://llvm.org/doxygen/structllvm_1_1DenseMapInfo_3_01std_1_1tuple_3_01Ts_8_8_8_01_4_01_4.html